### PR TITLE
2021-12-14-secure-jtag-on-imx-soc.md

### PR DESCRIPTION
**Pull request by kotylamichal**
: Fixes after Piotr review

### CODE REVIEW

**Review by artur-rs — APPROVED** [review]

**Review by pietrushnic — CHANGES_REQUESTED** [review]
Most of the problems were resolved. Thank you. There are some small remaining things.

Practice makes perfect.

**Comment by pietrushnic** [comment]
@artur-rs @kotylamichal there are still minor correction to apply.

File: blog/content/post/2021-12-14-secure-jtag-on-imx-soc.md [hunk 1]
```diff
@@ -25,67 +25,119 @@ categories:
...
+fields. With that tool we can burn fuses using their names - it is more safety
+than counting banks and words.
+
+- **JTAG compatible with OpenOCD**: On 
+[NXP sites](https://www.nxp.com/docs/en/application-note/AN4686.pdf) we can read
+that Secure JTAG is supported only by the Lauterbach environment and ARM-DS5 IDE
+with DSTREAM debugger. In 3mdeb we are always trying to use open-source
+software. Unfortunately, OpenOCD doesn't officially support Secure JTAG in i.MX
+SoC's, but we found not accepted yet patch from 2014 which add support for work
+with SJC (System JTAG controller) in a secure mode. Required patch work on
```
[the remark this answers is not in the document]
Comment (kotylamichal) [reply blog/content/post/2021-12-14-secure-jtag-on-imx-soc.md:63]: Added

File: blog/content/post/2021-12-14-secure-jtag-on-imx-soc.md [hunk 2]
```diff
@@ -25,67 +25,119 @@ categories:
...
-solution on i.MX6 but this same feature is available on a lot more NXP
-platforms. Unfortunately they have missing public documentation.
+development. It also may be helpful for adversaries while performing a
+reverse-engineering process - e.g. is a chance to dump the memory of your
+firmware and get access to strictly confidential information. In this article,
+we show how to lock JTAG access for users who don't have a special key. We will
+use Secure JTAG feature which is implemented on i.MX SoC's. In our lab, we
+tested Secure JTAG settings on i.MX6 but this same feature is available on a lot
+more NXP platforms. Unfortunately, some part of NXP documentation is not
+publicly available.
```
[the remark this answers is not in the document]
Comment (kotylamichal) [reply blog/content/post/2021-12-14-secure-jtag-on-imx-soc.md:35]: Fixed some misspells, I don't see more non-false errors (I use free version of Grammarly)

File: blog/content/post/2021-12-14-secure-jtag-on-imx-soc.md [hunk 3]
```diff
@@ -25,67 +25,119 @@ categories:
...
+if test "${gcc_werror}" = yes; then
+  GCC_WARNINGS="${GCC_WARNINGS} -Werror"
+fi
+```
+to:
+
+```bash
+# set default gcc warnings
+GCC_WARNINGS=""
+```
```
[the remark this answers is not in the document]
Comment (kotylamichal) [reply blog/content/post/2021-12-14-secure-jtag-on-imx-soc.md:105]: Ok

File: blog/content/post/2021-12-14-secure-jtag-on-imx-soc.md [hunk 4]
```diff
@@ -25,67 +25,119 @@ categories:
...
+OpenOCD from [official repository](git://git.code.sf.net/p/openocd/code) if you
+checkout code with `970a12aef` hash. There is a chance to apply patch in the
+actual version of OpenOCD, but it needs some code changes. As a hardware, we
+used [ARM-USB-OCD-H](https://www.olimex.com/Products/ARM/JTAG/ARM-USB-OCD-H/)
+(54,95 EUR) from Olimex.
+
+The simpliest way to build OpenOCD with SJC patch is use these commands:
+
+```shell
+$ git clone -n git://git.code.sf.net/p/openocd/code openocd-code
```
[the remark this answers is not in the document]
Comment (kotylamichal) [reply blog/content/post/2021-12-14-secure-jtag-on-imx-soc.md:73]: @pietrushnic Our fork is to do before publishing this article, or it is an idea for the future?  If we found a way to deal with SJC from OpenOCD we can publish a new, our patch that allows working with more than i.MX6. This also can be an idea for a new article (tested other i.MX devices, new patch, our fork with applied patch)

**Review by pietrushnic — APPROVED** [review]

**Pull request merged**